src/main/kotlin/com/example/ajaxproject/service/GroupChatServiceImpl.kt

```diff
@@ -0,0 +1,126 @@
+package com.example.ajaxproject.service
+
+import com.example.ajaxproject.dto.request.CreateChatDto
+import com.example.ajaxproject.dto.request.GroupChatDTO
+import com.example.ajaxproject.dto.responce.GroupChatMessageResponse
+import com.example.ajaxproject.exeption.NotFoundException
+import com.example.ajaxproject.exeption.WrongActionException
+import com.example.ajaxproject.model.GroupChatMessage
+import com.example.ajaxproject.model.GroupChatRoom
+import com.example.ajaxproject.model.User
+import com.example.ajaxproject.repository.GroupChatMessageRepository
+import com.example.ajaxproject.repository.GroupChatRoomRepository
+import com.example.ajaxproject.repository.UserRepository
+import com.example.ajaxproject.service.interfaces.GroupChatService
+import com.example.ajaxproject.service.mapper.GroupChatMessageMapper
+import org.bson.types.ObjectId
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Service
+
+@Service
+class GroupChatServiceImpl @Autowired constructor(
+    private val groupChatRoomRepository: GroupChatRoomRepository,
+    private val userRepository: UserRepository,
+    private val groupChatMessageRepository: GroupChatMessageRepository,
+    private val groupChatMessageMapper: GroupChatMessageMapper
+) : GroupChatService {
+
+    private val logger: Logger = LoggerFactory.getLogger(GroupChatServiceImpl::class.java)
+
+    override fun createGroupRoom(createChatDto: CreateChatDto): GroupChatRoom {
+
+        val user = userRepository.findById(createChatDto.adminId).orElseThrow {
+            logger.error("Error creating group chat room")
+            NotFoundException("User not found")
+        }
+
+        val groupChatRoom = GroupChatRoom(
+            id = ObjectId(),
+            chatName = createChatDto.chatName,
+            adminId = createChatDto.adminId,
+            chatMembers = listOf(user)
+        )
+
+        ArrayList(groupChatRoom.chatMembers).add(user)
+
+        logger.info("Group chat room created successfully")
```

Comment:
At this moment the room is not created. It will be created on `#save`, maybe.

Reply:
Yes , it's true) Fixed it